python_modules/dagster/dagster/_core/storage/local_compute_log_manager.py

```diff
@@ -226,9 +227,10 @@ def get_captured_local_path(self, log_key: Sequence[str], extension: str, partia
             filename = f"{filename}.partial"
         if len(filename) > MAX_FILENAME_LENGTH:
             filename = "{}.{}".format(non_secure_md5_hash_str(filebase.encode("utf-8")), extension)
-        location = os.path.join(self._base_dir, *namespace, filename)
-        location = os.path.abspath(location)
-        if not location.startswith(self._base_dir):
+        baseDirLocation = Path(self._base_dir).resolve()
+        location = baseDirLocation.joinpath(*namespace, filename)
+        location = location.resolve()
```

Comment:
this changes the type of the return value which breaks things downstream 
https://buildkite.com/dagster/unit-tests/builds/354